### PR TITLE
Operator app

### PR DESCRIPTION
**Pull request by janmazak**

### CODE REVIEW

File: src/crypto-providers/ledgerCryptoProvider.ts [hunk 1]
```diff
@@ -19,6 +19,7 @@ import {
   TxCertificateKeys,
   _Withdrawal,
   _StakepoolRegistrationCert,
+  _StakepoolRetirementCert, // I don't like importing tons of stuff, and in a seemingly random order, what about using something like tx.StakepoolRetirementCert?
```
Comment (DavidTranDucVL) [inline src/crypto-providers/ledgerCryptoProvider.ts:22]: I don't mind either. We can do that.

File: src/crypto-providers/ledgerCryptoProvider.ts [hunk 2]
```diff
@@ -258,6 +260,23 @@ export const LedgerCryptoProvider: () => Promise<CryptoProvider> = async () => {
     }
   }
 
+  // TODO we use both StakePool and Stakepool in names
```
Comment (DavidTranDucVL) [inline src/crypto-providers/ledgerCryptoProvider.ts:263]: Yeah, it would be nice to unify this, I'm more leaning to `Stakepool`

File: src/transaction/guards.ts [hunk 3]
```diff
@@ -113,6 +114,7 @@ const isMargin = (_value: any) => {
     && Number.isInteger(denominator)
 }
 
+// TODO should be renamed to isPoolMetadata to avoid potential confusion with tx metadata
```
Comment (DavidTranDucVL) [inline src/transaction/guards.ts:117]: I agree with this idea.

File: src/transaction/guards.ts [hunk 4]
```diff
@@ -148,6 +150,21 @@ export const isStakepoolRegistrationCert = (
...
+): value is TxStakepoolRetirementCert => {
+  if (!Array.isArray(value) || value.length !== 3) return false
+  const [
+    type,
+    poolKeyHash,
+    retirementEpoch,
+  ] = value
+  return type === TxCertificateKeys.STAKEPOOL_RETIREMENT
+    && Buffer.isBuffer(poolKeyHash)
+    && poolKeyHash.length === 28 // TODO not checked elsewhere; make 28 a named constant and perhaps have a function for checking fixed-length buffer
```
Comment (DavidTranDucVL) [inline src/transaction/guards.ts:164]: Yeah, I think we should have this, good idea.

**Comment by refi93** [comment]
bug:

```
cardano-hw-cli --tx-body-file tx.raw \
--hw-signing-file cold.hwsfile \
--hw-signing-file payment.hwsfile \
--testnet-magic 42 \
--out-file tx.signed
```
results in a transaction with a single witness - the tx being signed has a pool retirement certificate

This happens only if the cold.hwsfile comes before the payment.hwsfile, not the other way around

File: src/crypto-providers/ledgerCryptoProvider.ts [hunk 5]
```diff
@@ -309,25 +387,37 @@ export const LedgerCryptoProvider: () => Promise<CryptoProvider> = async () => {
...
+    ) {
+      throw Error(Errors.PoolRegistrationAsOperatorNotSupported)
+    }
   }
 
   const ledgerSignTx = async (
     txAux: _TxAux, signingFiles: HwSigningData[], network: Network, changeOutputFiles: HwSigningData[],
   ): Promise<LedgerWitness[]> => {
-    ensureFirmwareSupportsParams(txAux)
-    const { paymentSigningFiles, stakeSigningFiles } = filterSigningFiles(signingFiles)
```
Comment (DavidTranDucVL) [inline src/crypto-providers/ledgerCryptoProvider.ts:318]: Is there any reason why was this removed? It's breaking old logic and causing wrong paths mapping.

**Comment by DavidTranDucVL** [comment]
Merging this to make life easier with rebasing and working on voting app. We can open another PR for fixes.
